src/Umbraco.Tests/FrontEnd/UmbracoHelperTests.cs

```diff
@@ -78,5 +93,41 @@ public void Create_Encrypted_RouteString_From_Dictionary()
             
             Assert.AreEqual(expectedResult, result);
         }
+
+        [Test]
+        public void Truncate_By_Words()
+        {
+            var text = "Hello world, this is some text <a href='blah'>with a link</a>";
+
+            var helper = new UmbracoHelper();
+
+            var result = helper.TruncateByWords(text, 4).ToString();
+
+            Assert.AreEqual("Hello world, this is&hellip;", result);
+        }
+
+        [Test]
+        public void Truncate_By_Words_With_Tag()
+        {
+            var text = "Hello world, <b>this</b> is some text <a href='blah'>with a link</a>";
```

Comment:
Ah, I guess we already went down this rabbithole with the truncate method for number of characters.. which is also already a pretty complex method. Yikes.